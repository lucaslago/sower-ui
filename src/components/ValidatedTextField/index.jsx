import React, { Component } from 'react';
import TextField from 'material-ui/TextField';

export default class ValidatedTextField extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fieldValue: ''
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.handleChange(event.target.value);
    this.setState({ fieldValue: event.target.value });
  }

  render() {
    return (
      <TextField
        floatingLabelText={ this.props.label }
        type={ this.props.type }
        fullWidth={ true }
        disabled={ this.props.disabled }
        errorText={ this.props.showValidationError ? this.props.errorText : '' }
        value={ this.state.fieldValue }
        onChange={ this.handleChange }
        className="validated-text-field"
      />
    );
  }
}
