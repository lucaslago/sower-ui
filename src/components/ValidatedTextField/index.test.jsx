import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import ValidatedTextField from './index';
import TextField from 'material-ui/TextField';

describe('<ValidatedTextField />', () => {
  const fieldSelector = '.validated-text-field';
  const handleChangeFake = () => {};

  it('renders without crashing', () => {
    const wrapper = shallow(<ValidatedTextField label="login" />);
    expect(wrapper.find(fieldSelector).length).toEqual(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should not display error when showValidationError is false', () => {
    const wrapper = shallow(
      <ValidatedTextField
        label="login"
        type="text"
        errorText="Field required"
        showValidationError={ false }
        handleChange = { handleChangeFake }
      />
    );
    expect(wrapper.find(fieldSelector).text()).not.toContain('Field required');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should display error when showValidationError is true', () => {
    const wrapper = render(
      <MuiThemeProvider>
        <ValidatedTextField
          label="login"
          type="text"
          errorText="Field required"
          showValidationError={ true }
          handleChange = { handleChangeFake }
        />
      </MuiThemeProvider>
    );
    expect(wrapper.find(fieldSelector).text()).toContain('Field required');
  });
});
