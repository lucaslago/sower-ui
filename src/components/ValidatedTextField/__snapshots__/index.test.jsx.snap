exports[`<ValidatedTextField /> renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TextField
    className="validated-text-field"
    disabled={false}
    errorText=""
    floatingLabelFixed={false}
    floatingLabelText="login"
    fullWidth={true}
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value="" />,
  "nodes": Array [
    <TextField
      className="validated-text-field"
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="login"
      fullWidth={true}
      multiLine={false}
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="" />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ValidatedTextField
        label="login" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ValidatedTextField {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "props": Object {
          "label": "login",
        },
        "refs": Object {},
        "state": Object {
          "fieldValue": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TextField
          className="validated-text-field"
          disabled={false}
          errorText=""
          floatingLabelFixed={false}
          floatingLabelText="login"
          fullWidth={true}
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
          value="" />,
        "_debugID": 2,
        "_renderedOutput": <TextField
          className="validated-text-field"
          disabled={false}
          errorText=""
          floatingLabelFixed={false}
          floatingLabelText="login"
          fullWidth={true}
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
          value="" />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ValidatedTextField
    label="login" />,
}
`;

exports[`<ValidatedTextField /> should not display error when showValidationError is false 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TextField
    className="validated-text-field"
    disabled={false}
    errorText=""
    floatingLabelFixed={false}
    floatingLabelText="login"
    fullWidth={true}
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value="" />,
  "nodes": Array [
    <TextField
      className="validated-text-field"
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="login"
      fullWidth={true}
      multiLine={false}
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="" />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ValidatedTextField
        errorText="Field required"
        handleChange={[Function]}
        label="login"
        showValidationError={false}
        type="text" />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ValidatedTextField {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "props": Object {
          "errorText": "Field required",
          "handleChange": [Function],
          "label": "login",
          "showValidationError": false,
          "type": "text",
        },
        "refs": Object {},
        "state": Object {
          "fieldValue": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TextField
          className="validated-text-field"
          disabled={false}
          errorText=""
          floatingLabelFixed={false}
          floatingLabelText="login"
          fullWidth={true}
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
          value="" />,
        "_debugID": 4,
        "_renderedOutput": <TextField
          className="validated-text-field"
          disabled={false}
          errorText=""
          floatingLabelFixed={false}
          floatingLabelText="login"
          fullWidth={true}
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
          value="" />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ValidatedTextField
    errorText="Field required"
    handleChange={[Function]}
    label="login"
    showValidationError={false}
    type="text" />,
}
`;
